adapters/amqp-vertx-base/src/test/java/org/eclipse/hono/adapter/amqp/VertxBasedAmqpProtocolAdapterTest.java

```diff
@@ -1291,6 +1303,121 @@ public void testConnectionFailsForUnauthenticatedDeviceIfAdapterLevelConnectionL
+    }
+
+    /**
+     * Verifies that the adapter closes the connection to an authenticated device when a notification
+     * has been received that the tenant of the device has been disabled.
+     */
+    @Test
+    public void testDeviceConnectionIsClosedOnTenantDisabledNotification() {
+
+        testDeviceConnectionIsClosedOnDeviceOrTenantChangeNotification( new TenantChangeNotification(
+                LifecycleChange.UPDATE, TEST_TENANT_ID, Instant.now(), false));
+    }
+
+    /**
+     * Verifies that the adapter closes the connection to an authenticated device when a notification
+     * about the deletion of all device data of the tenant of that device has been received.
+     */
+    @Test
+    public void testDeviceConnectionIsClosedOnAllDevicesOfTenantDeletedNotification() {
+
+        testDeviceConnectionIsClosedOnDeviceOrTenantChangeNotification(
+                new AllDevicesOfTenantDeletedNotification(TEST_TENANT_ID, Instant.now()));
+    }
+
+    private <T extends AbstractNotification> void testDeviceConnectionIsClosedOnDeviceOrTenantChangeNotification(
+            final T notification) {
+
+        // GIVEN an AMQP adapter
+        givenAnAdapter(properties);
+
+        final Promise<Void> startPromise = Promise.promise();
+        adapter.doStart(startPromise);
+        assertThat(startPromise.future().succeeded()).isTrue();
+
+        @SuppressWarnings("unchecked")
+        final ArgumentCaptor<Handler<io.vertx.core.eventbus.Message<T>>> notificationHandlerCaptor = getEventBusConsumerHandlerArgumentCaptor(
+                (NotificationType<T>) notification.getType());
+
+        // with an enabled tenant
+        givenAConfiguredTenant(TEST_TENANT_ID, true);
+
+        // WHEN a device connects
+        final Device authenticatedDevice = new Device(TEST_TENANT_ID, TEST_DEVICE);
+        final Record record = new RecordImpl();
+        record.set(AmqpAdapterConstants.KEY_CLIENT_DEVICE, Device.class, authenticatedDevice);
+        final ProtonConnection deviceConnection = mock(ProtonConnection.class);
+        when(deviceConnection.attachments()).thenReturn(record);
+        when(deviceConnection.getRemoteContainer()).thenReturn("deviceContainer");
+        adapter.onConnectRequest(deviceConnection);
+        final ArgumentCaptor<Handler<AsyncResult<ProtonConnection>>> openHandler = VertxMockSupport.argumentCaptorHandler();
+        verify(deviceConnection).openHandler(openHandler.capture());
+        openHandler.getValue().handle(Future.succeededFuture(deviceConnection));
+
+        // THEN the connection count is incremented
+        verify(metrics).incrementConnections(TEST_TENANT_ID);
+
+        // AND WHEN a notification is sent about the tenant/device having been deleted or disabled
+        sendViaEventBusMock(notification, notificationHandlerCaptor.getValue());
+
+        // THEN the device connection is closed
+        verify(deviceConnection).close();
```

Comment:
how about letting the device also create a command consumer and the check that command consumer is getting closed as well?